scripts/models.py

```diff
@@ -0,0 +1,244 @@
+    unitText: str = None
+    value: Union[bool, float, str, int, List[Union[bool, float, str, int]]] = None
+    valueReference: PropertyValue = None
+    propertyID: Union[str, AnyUrl] = None
+
+
+PropertyValue.update_forward_refs()
+
+
+class Identifier(BaseModel):
+    identifier: Union[str, AnyUrl, PropertyValue]
+
+
+class Contributor(BaseModel):
+    identifier: Identifier = None
+    name: str
+    email: EmailStr = None
+    url: AnyUrl = None
+    roleName: List[RoleType]
+    includeInCitation: bool = True
+    awardNumber: str = None
+
+
+class Person(Contributor):
+    affiliation: List[str]
+
+
+class Organization(Contributor):
+    contactPoint: str
+
+
+class EthicsApproval(BaseModel):
+    identifier: Union[str, AnyUrl, Identifier] = None
+    name: str
+    url: str
+
+
+class Resource(BaseModel):
+    identifier: Identifier = None
+    name: str = None
+    url: str
+    repository: Union[str, AnyUrl] = None
+    relation: Relation
+
+
+class About(BaseModel):
+    identifier: Union[str, AnyUrl, Identifier] = None
+    name: str = None
+
+
+class AccessRequirements(BaseModel):
+    status: AccessType
+    email: EmailStr = None
+    contactPoint: str = None
+    description: str = None
+    embargoedUntil: date = None
+
+
+class DandisetStat(BaseModel):
+    numberOfFiles: int = None
+    numberOfSubjects: int = None
+    numberOfSamples: int = None
+    numberOfCells: int = None
+    dataStandard: List[str] = None
+    modality: List[str] = None
+
+
+class Digest(BaseModel):
+    value: str
+    cryptoType: AnyUrl
+
+
+class BioSample(BaseModel):
+    assayType: AnyUrl
+    anatomy: AnyUrl
+
+
+class BaseDandiset(BaseModel):
+    name: str
+    description: str
+
+
+class Dandiset(BaseDandiset):
+    """A body of structured information describing a DANDI dataset
+    """
+    schemaVersion: str = Field(default="0.0.0", readonly=True)
+    identifier: Identifier = Field(readonly=True)
+
+    contributor: List[Union[Person, Organization]]
+
+    license: License
+    keywords: List[str]
+    access: List[AccessRequirements]
+    about: List[About] = None
+    study_target: List[Union[str, AnyUrl]] = None
+    protocol: List[str] = None
+    ethicsApproval: List[EthicsApproval] = None
+    relatedResource: List[Resource] = None
+    acknowledgement: str = None
+
+    # From assets
+    measurementTechnique: List[str] = Field(readonly=True)
+    variableMeasured: List[PropertyValue] = Field(readonly=True)
+    dandisetStats: DandisetStat = Field(readonly=True)
+
+
+class PublishedDandiset(Dandiset):
+    # On publish
+    version: str = Field(readonly=True)
+    datePublished: date = Field(readonly=True)
+    url: AnyUrl = Field(readonly=True)
+    contentSize: str = Field(readonly=True)
+    repository: AnyUrl = Field(readonly=True)
+    manifestLocation: Union[AnyUrl, List[AnyUrl]] = Field(readonly=True)
+    generatedBy: Optional[AnyUrl] = Field(None, readonly=True)
+
+
+class Asset(BaseModel):
```

Comment:
@yarikoptic  and @mgrauer - please take a look at this asset schema to see if there are any critical things missing at this point. (also would be good to review the dandiset schemas above)

also i know how to do some of the complicated constraints in json schema, but not how to encode them in pydantic yet. so that specific update will also happen. 

@AlmightyYakob - i think the current json schema included in this branch for the dandiset would be thing to update in relation to the UI.